examples/air-quality-sensor-app/silabs/src/SensorManager.cpp

```diff
@@ -0,0 +1,180 @@
+{
+    AirQualitySensorManager::InitInstance();
+}
+
+CHIP_ERROR SensorManager::Init()
+{
+    DeviceLayer::PlatformMgr().ScheduleWork(InitAirQualitySensorManager);
+    // Create cmsisos sw timer for air quality sensor timer.
+    mSensorTimer = osTimerNew(SensorTimerEventHandler, osTimerPeriodic, nullptr, nullptr);
+    if (mSensorTimer == NULL)
+    {
+        ChipLogDetail(AppServer, "mSensorTimer timer create failed");
+        return APP_ERROR_CREATE_TIMER_FAILED;
+    }
+
+#ifdef USE_AIR_QUALITY_SENSOR
+    if (SL_STATUS_OK != AirQualitySensor::Init())
+    {
+        ChipLogDetail(AppServer, "Failed to Init Sensor");
+        return CHIP_ERROR_INTERNAL;
+    }
+#endif
+    // Update Air Quality immediatly at bootup
+    SensorTimerEventHandler(nullptr);
+    // Trigger periodic update
+    uint32_t delayTicks = ((uint64_t) osKernelGetTickFreq() * kSensorTImerPeriodMs) / 1000;
+
+    // Starts or restarts the function timer
+    if (osTimerStart(mSensorTimer, delayTicks))
+    {
+        ChipLogDetail(AppServer, "mSensor Timer start() failed");
+        appError(APP_ERROR_START_TIMER_FAILED);
+    }
+    return CHIP_NO_ERROR;
+}
+
+void writeAirQualityToAttribute(intptr_t context)
+{
+    int32_t * air_quality_ptr = reinterpret_cast<int32_t *>(context);
+    AirQualitySensorManager::GetInstance()->OnAirQualityChangeHandler(classifyAirQuality(*air_quality_ptr));
+    ChipLogDetail(AppServer, "RAW AirQuality value: %ld and corresponding Enum value : %d", *air_quality_ptr,
+                  chip::to_underlying(AirQualitySensorManager::GetInstance()->GetAirQuality()));
+    AppTask::GetAppTask().UpdateAirQualitySensorUI();
+}
+
+void SensorManager::SensorTimerEventHandler(void * arg)
+{
+    int32_t air_quality;
+#ifdef USE_AIR_QUALITY_SENSOR
+    if (SL_STATUS_OK != AirQualitySensor::GetAirQuality(air_quality))
+    {
+        ChipLogDetail(AppServer, "Failed to read Air Quality !!!");
+        return;
+    }
+#else
+    // Initialize static variables to keep track of the current index and repetition count
+    static uint8_t nbOfRepetition = 0;
+    static uint8_t simulatedIndex = 0;
+
+    // Ensure the simulatedIndex wraps around the array size to avoid out-of-bounds access
+    simulatedIndex = simulatedIndex % ArraySize(mSimulatedAirQuality);
+    // Retrieve the current air quality value from the simulated data array using the simulatedIndex
+    air_quality = mSimulatedAirQuality[simulatedIndex];
+
+    // Increment the repetition count
+    nbOfRepetition++;
+    // Check if the number of repetitions has reached the threshold to simulate a new reading
+    if (nbOfRepetition >= kSimulatedReadingFrequency)
+    {
+        // Move to the next index for the next simulated reading
+        simulatedIndex++;
+        // Reset the repetition count
+        nbOfRepetition = 0;
+    }
+#endif // USE_AIR_QUALITY_SENSOR
+    // create pointer for the int32_t air_quality
+    int32_t * air_quality_ptr = new int32_t(air_quality);
```

Comment:
memory leak. I don't see you freeing it.

Reply:
they are deleting in the function `writeAirQualityToAttribute`.